app/gameengine.py

```diff
@@ -3395,6 +3438,16 @@ def archive_hand_continuation():
                 prevent_life_loss = effect["prevent_life_loss"]
                 self.deal_damage(source_player, target_player, dealing_card, target_card, amount, special, prevent_life_loss, {}, self.continue_resolving_effects)
                 passed_on_continuation = True
+            case EffectType.EffectType_DealDamagePerStacked:
+                match effect.get("limitation"):
+                    case "center":
+                        holomems = effect_player.center
+                    case _:
+                        holomems = effect_player.get_holomem_on_stage()
+                num_of_stacked_cards = len([card for holomem in holomems for card in holomem["stacked_cards"] if is_card_holomem(card)])
+                effect["amount"] *= num_of_stacked_cards
+                effect["effect_type"] = EffectType.EffectType_DealDamage
+                self.do_effect(effect_player, effect)
```

Comment:
Don't call do_effect directly because continuations might not work right. Instead, prepend this to the front of the effects to resolve.
Otherwise, this do_effect you're calling could kick off a continuation, but then this code and the continuation will both want to continue resolving effects in the effect loop.